src/parser.cpp

```diff
@@ -158,7 +156,7 @@ namespace ada::parser {
               url.update_base_pathname(base_url->get_pathname());
               url.update_base_search(base_url->get_search());
             }
-
+            url.update_unencoded_base_hash(*fragment);
```

Comment:
I'm kind of suspicious about the naming of this function, since the naming convention is not common. I don't have an alternative tho.

Reply:
This PR does not introduce a new name.

Reply:
Yes, I was commenting on it for future reference.